runtime/runtime/src/lib.rs

```diff
@@ -1295,6 +1301,36 @@ impl Runtime {
         Ok(())
     }
 
+    #[cfg(feature = "ganache")]
+    fn apply_state_patches(
+        &self,
+        state_update: &mut TrieUpdate,
+        states_to_patch: Vec<StateRecord>,
+    ) {
+        for record in states_to_patch {
+            match record {
+                StateRecord::Account { account_id, account } => {
+                    set_account(state_update, account_id, &account);
+                }
+                StateRecord::Data { account_id, data_key, value } => {
+                    state_update.set(TrieKey::ContractData { key: data_key, account_id }, value);
+                }
+                StateRecord::Contract { account_id, code } => {
+                    let acc = get_account(&state_update, &account_id).expect("Failed to read state").expect("Code state record should be preceded by the corresponding account record");
+                    // Recompute contract code hash.
+                    let code = ContractCode::new(code, None);
+                    set_code(state_update, account_id, &code);
+                    assert_eq!(code.get_hash(), acc.code_hash());
+                }
+                StateRecord::AccessKey { account_id, public_key, access_key } => {
+                    set_access_key(state_update, account_id, public_key, &access_key);
+                }
+                _ => unimplemented!("patch_state can only patch Account, AccessKey, Contract and Data kind of StateRecord")
+            }
+        }
+        state_update.commit(StateChangeCause::PatchState);
```

Comment:
@bowenwang1996 Although we may not need StateChangeCauseView, but we seem to need a reason for state_update.commit here, and that we need to add StateChangeCause::PatchState. Given StateChangeCause::PatchState need to be added in primitives, adding StateChangeCauseView::PatchState doesn't hurt

Reply:
@ailisp it actually hurts when you want to have an exhaustive match for all the possible enum variants and now you have to deal with the case that cannot happen in production. Please, hide it behind a feature flag. Alternatively, I can suggest we abuse some other patch reason (`Migration` or `InitialState`).

Reply:
Ah sorry I didn't see this comment. I could not figure out the feature magic to make `cargo check` and `cargo test --workspace --all-features` work at the same time, on the earlier commit where StateChangeCause::PatchState is under a feature. I'll abuse Migration reason